### ISSUE TITLE
Coach view: update ability to view independent practice quiz reports

### ISSUE THREAD

**Issue by sairina** [learningequality/kolibri#8518]
### User stories for this issue:

1. As a coach, I want to see independent practice quizzes in the lesson resource list of a lesson under the tab "Reports", so that I can access the report for those independent practice quizzes. 
2. As a coach, I want to see progress in the table updated next to an independent practice quiz's learner list inside of a lesson, so that I can get an overview of how many learners have engaged with the independent practice quiz.
3. As a coach, I want to see a list of all learners who have been assigned the independent practice quiz when I click the title of the independent practice quiz, so that I can get an overview of how learners have performed on this independent practice quiz.
4. As a coach, I want to view "difficult questions" from the most recent attempt my learners have made so that I can view that in one space.
5. As a coach, I want to see a preview of the independent practice quiz I have selected if I click on the "preview" button.

|Frame|Explanation|
|--|--|
|![Screen Shot 2021-10-21 at 12 01 34 PM](https://user-images.githubusercontent.com/13563002/138345302-2f39a8ed-31cb-4000-b640-a14b09a65e8c.png)| From Reports > Lessons subtab |
|![Screen Shot 2021-10-21 at 12 01 55 PM](https://user-images.githubusercontent.com/13563002/138345303-604d75f2-e783-41ea-be43-819bcdc17f2e.png)| When coach clicks on a lesson title, they are taken to the lesson resource list. In the report subtab, coach should see an independent practice quiz with progress noted |
|![Screen Shot 2021-10-21 at 12 02 22 PM](https://user-images.githubusercontent.com/13563002/138345307-db122500-dac2-43bb-8b6f-6f81a0df85c1.png)| When a coach clicks into the independent practice quiz, this is the view with some new metadata, and the table looks different than that of a normally assigned quiz |
|![Screen Shot 2021-10-21 at 12 02 29 PM](https://user-images.githubusercontent.com/13563002/138345308-212c5fd4-0466-44f5-ae42-a98e0c9f5e5b.png)| When a coach clicks on the Difficult questions subtab, this is the view. Note that the questions that appear should be from the MOST RECENT attempt for all users only.|
|![Screen Shot 2021-10-21 at 11 40 13 AM](https://user-images.githubusercontent.com/13563002/138346377-bf0f5b01-6e81-4a1a-b00a-0e66543cbb15.png)|This is a view of the preview of the independent practice quiz if a coach clicks on the "Preview" button|

### Notes and changes to current designs:
- [Figma for viewing independent practice quiz reports](https://www.figma.com/file/TfaSTT9gs6fPKhGdNUjffU/Channel-quizzes?node-id=1013%3A10037)
- OUT OF SCOPE: Creating the learner reports for when a coach clicks on a learner's name or difficult question report
- Routing and navigation already exists for this user story, so you'll just need to connect the `ReportsPracticeQuizQuestionPage` (difficult questions component) and `ReportsLessonPracticeQuizLearnerPage` (individual learner report) to the dummy pages in the route.
- All strings required should already be in `commonCoreStrings`, `commonCoachStrings`, or `commonLearnStrings`
- For the 3rd user story, there are some changes to be noted
   - Connect "Preview" functionality as **button**
      - "Preview" should be rendered the same as when the coach assigns the channel quiz to the learner
   - In the **reports table for learners**, add "Latest score," "Attempts," columns instead of "Time spent" from exercises
      - Percentage change shows improvement or regression since previous attempt
      
### Criteria for success:
**Please copy these into your PR for "Steps for Testing"!**

_This issue is done when the checkbox and Gherkin stories below work_

- [ ] clicking the "Preview" button from Reports > Lessons > (click on Lesson title) > "Preview" button should take me to a preview of the quiz
- [ ] coach can view lesson resource list in reports after a learner engages with independent practice quiz
```
Feature: Independent practice quizzes assigned in-lesson have reports
	Scenario: coach views lesson resource list in reports after a learner engages with independent practice quiz
		Given that I have a lesson assigned to learners
			And there is an independent practice quiz in the lesson
			And I go into the Reports tab
			And I click into the Lessons subtab
			And I click into a Lesson
		Then I see the independent practice quiz listed in the lesson resource list
```
- [ ] coach can view independent practice quiz report after a learner engages with independent practice quiz
```
Feature: Independent practice quizzes assigned in-lesson have reports
	Scenario: coach views independent practice quiz report after a learner engages with independent practice quiz
		Given that I have a lesson assigned to learners
			And there is an independent practice quiz in the lesson
			And learners have started to engage with the independent practice quiz
		When at least one learner has submitted the quiz
			And I click on the lesson report 
		Then I see that there is progress in the 'Progress' column of the lesson report

```
- [ ] coach can view independent practice quiz report after learner engages with independent practice quiz
```
Feature: Independent practice quizzes assigned in-lesson have reports
	Scenario: coach views independent practice quiz report after learner engages with independent practice quiz
		Given that I have a lesson assigned to learners
			And there is an independent practice quiz in the lesson
			And learners have started to engage with the independent practice quiz
		When I click into the independent practice quiz report in the lesson report
		Then I see that individual learner progress on the independent practice quiz in a table
			And I see a subtab with difficult questions
			And I see a button that has a preview of the independent practice quiz
			And I see a column in the table with latest score and attempts
```
- [ ] coach can see the correct difficult questions for each attempt
```
Feature: Difficult question reports for independent practice quizzes from lessons
Scenario: coach sees the correct difficult questions for most recent attempt
		Given that I am in 'Reports'
			And I click on the 'Lessons' subtab
			And I click on a lesson title
			And I click on the title of an independent practice quiz in the 'Reports' subtab
		When I click on the 'Difficult questions' subtab
		Then I see the correct difficult questions for most recent attempt
```

**Comment by bjester** [comment]
Hey team! Please [add your planning poker estimate](https://app.zenhub.com/workspaces/Kolibri-Ecosystem-6047e525b8b9d900182e2d6f/issues/learningequality/kolibri/8518?planning-poker) with ZenHub @jredrejo @marcellamaki @sairina @rtibbles @MisRob @nucleogenesis @jamalex

**Comment by rtibbles** [comment]
Fixed in #8521
